receiver/prometheusremotewritereceiver/receiver.go

```diff
@@ -150,8 +155,106 @@ func (prw *prometheusRemoteWriteReceiver) parseProto(contentType string) (promco
 }
 
 // translateV2 translates a v2 remote-write request into OTLP metrics.
-// For now translateV2 is not implemented and returns an empty metrics.
+// translate is not feature complete.
 // nolint
-func (prw *prometheusRemoteWriteReceiver) translateV2(_ context.Context, _ *writev2.Request) (pmetric.Metrics, promremote.WriteResponseStats, error) {
-	return pmetric.NewMetrics(), promremote.WriteResponseStats{}, nil
+func (prw *prometheusRemoteWriteReceiver) translateV2(_ context.Context, req *writev2.Request) (pmetric.Metrics, promremote.WriteResponseStats, error) {
+	var (
+		badRequestErrors error
+		otelMetrics      = pmetric.NewMetrics()
+		labelsBuilder    = labels.NewScratchBuilder(0)
+		stats            = promremote.WriteResponseStats{}
+	)
+
+	for _, ts := range req.Timeseries {
+		ls := ts.ToLabels(&labelsBuilder, req.Symbols)
+
+		if !ls.Has(labels.MetricName) {
+			badRequestErrors = errors.Join(badRequestErrors, fmt.Errorf("missing metric name in labels"))
+			continue
+		} else if duplicateLabel, hasDuplicate := ls.HasDuplicateLabelNames(); hasDuplicate {
+			badRequestErrors = errors.Join(badRequestErrors, fmt.Errorf("duplicate label %q in labels", duplicateLabel))
+			continue
+		}
+
+		var rm pmetric.ResourceMetrics
+		// This cache should be populated by the metric 'target_info', but we're not handling it yet.
+		hashedJobAndInstance := xxhash.Sum64String(ls.Get("job") + ls.Get("instance"))
```

Comment:
you will need to add a separator to ensure these aren't the same:

* `job=foo`, `instance=bar`
* `job=f`, `instance=oobar`

Prometheus uses
```golang
var seps = []byte{'\xff'}`
```